server/src/main/java/org/elasticsearch/search/fetch/FetchSubPhase.java

```diff
@@ -100,8 +99,8 @@ public IndexReader topLevelReader() {
     /**
      * Returns a {@link FetchSubPhaseProcessor} for this sub phase.
      *
-     * If nothing should be executed for the provided {@link SearchContext}, then the
+     * If nothing should be executed for the provided {@code FetchContext}, then the
      * implementation should return {@code null}
      */
-    FetchSubPhaseProcessor getProcessor(SearchContext searchContext, SearchLookup lookup) throws IOException;
+    FetchSubPhaseProcessor getProcessor(FetchContext fetchContext, SearchLookup lookup) throws IOException;
```

Comment:
Do you think `lookup` should be a member on `FetchContext`? I know I just added it, but now that you are proposing our own "context" I think I'd be nice to fold it in.

Reply:
Yes! I gave it a quick try after the merge but I think we really want to remove the QueryShardContext#newFetchLookup() method at the same time and it's not obvious to me how to do that yet, so I thought I'd leave it for a follow-up.

Reply:
I'd be ok moving to a member now if you're happy with it. It can wait too. Either way.

It isn't so much that we want to get rid of `newFetchLookup` as that we want it to return a `FetchLookup` instead of a `SearchLookup`. We just haven't boiled out the common interface for those.